diff --git a/src/internal/pthread_impl.h b/src/internal/pthread_impl.h
index f6a4f2c2..3c62c628 100644
--- a/src/internal/pthread_impl.h
+++ b/src/internal/pthread_impl.h
@@ -159,7 +159,7 @@ void __block_all_sigs(void *);
 void __block_app_sigs(void *);
 void __restore_sigs(void *);
 
-#define DEFAULT_STACK_SIZE 81920
+#define DEFAULT_STACK_SIZE 8388608
 #define DEFAULT_GUARD_SIZE 4096
 
 #define __ATTRP_C11_THREAD ((void*)(uintptr_t)-1)
